testTruckRender.gd

```diff
@@ -0,0 +1,56 @@
+@tool
+extends Sprite2D
+
+var truckModel;
+var postProcObject;
+var lastPos: Vector2;
+var currentPos: Vector2;
+var extend_vec: Vector2;
+
+var angle_to = 0;
+
+var lastTick = 0
+
+@export var modelTint: Color = Color.RED:
+	set(val):
+		modelTint = val;
+@export_range(0, 1, 0.1) var modelTintFactor:
+	set(val):
+		modelTintFactor = val;
+		
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	truckModel = $SubViewport/testTruck
+	postProcObject = get_node("SubViewport/Camera3D/Postprocess")
+	pass # Replace with function body.
+
+func _draw():
+#	if !Engine.is_editor_hint():
+		draw_line(position, position + (extend_vec), Color.RED, 2);
+		draw_arc(position, 20, 0, deg_to_rad(-1 * angle_to), 20, Color.YELLOW, 2)
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
+	if postProcObject:
+		postProcObject.get_surface_override_material(0).set_shader_parameter("tint_color", modelTint)
+		postProcObject.get_surface_override_material(0).set_shader_parameter("tint_effect_factor", modelTintFactor)
+	if !Engine.is_editor_hint():
+		var controlNode = get_tree().get_current_scene().get_node('ControlSet')
+		if controlNode.get_node('VBoxContainer/PauseUnpause') and controlNode.get_node("VBoxContainer/PauseUnpause").isPaused:
+			return
+		if controlNode.get_node("VBoxContainer/HScrollBar") and controlNode.get_node("VBoxContainer/HScrollBar").value <= 0.01:
+			return
+	
+	if get_tree().get_current_scene():
```

Comment:
What is this checking?

Reply:
This (and the similar checks above it) exit out of the process loop early if the controls are set to paused, or if the scrollbar is set low enough that it should be zero. Essentially stops the lastPos/currentPos and rotation etc. from being updated while the thing is actually not moving.

Reply:
The check on line 44 stops errors I got from viewing the truck's scene else we get errors since it attempts to grab a position from the SubViewport node rather than the Sprite2D for some reason.